### PR TITLE
fix edge case where fkey is 0 in relation reference widget 

### PR DESCRIPTION
**Pull request by 3nids**
if a not null constraint was set, the form could not be validated.

### CODE REVIEW

File: src/native/mac/qgsmacnative.mm [hunk 1]
```diff
@@ -60,7 +61,10 @@ - ( BOOL )userNotificationCenter:( NSUserNotificationCenter * )center shouldPres
 
 void QgsMacNative::setIconPath( const QString &iconPath )
 {
-  mQgsUserNotificationCenter->_qgisIcon = QtMac::toNSImage( QPixmap( iconPath ) );
+  QImage image = QPixmap( iconPath ).toImage();
```
Comment (nyalldawson) [inline src/native/mac/qgsmacnative.mm:64]: unrelated change

Comment (3nids) [reply src/native/mac/qgsmacnative.mm]: thanks

**Review by nyalldawson — APPROVED** [review]
